### ISSUE TITLE
Can't delete files in NTFS filesystem mounted on Linux Mint / Ubuntu

### ISSUE THREAD

**Issue by millerbarros** [adobe/brackets#10581]
Hello, I am a user in Windows brackets (where the brackets worked normally) at a time, but today I'm using **Linux Mint 17.1** and before that I used **Ubuntu 14.10**. I have a problem that may be related to the OS since Linux Mint is derived from Ubuntu and this problem is repeated in both.

Everytime I try to delete a file using the sidebar I get the following error message:

**"An error occurred When trying to delete the file /path/to/file.html. (Unknown error)"**

Has anyone had this problem? If so, how can I fix?

**Comment by dvorapa** [comment]
Can you delete it in Nautilus? If you can't, try to change permissions for that file from root to user's.

**Comment by millerbarros** [comment]
Yes, I can delete using Nautilus. When I need to delete a file, I right-click on it, select "Show in OS" and I delete the file in Nautilus, but if I try to select delete both files and folders the Brackets show this error message.
Also, I can rename folders and files with the sidebar, so I know that is not a problem with user permission.

**Comment by millerbarros** [comment]
Someone can help me with this problem?

**Comment by dvorapa** [comment]
I really think it is an issue caused by permissions. Have you tried to set `chmod` access openned (like `-R 777` or so)? Have you tried to reinstall Brackets? Have you tried to run Brackets with root access? Maybe also broken OS installation... I don't think it could be brackets-shell bug (but maybe I am wrong about this).

**Comment by peterflynn** [comment]
@millerbarros So you're able to rename a file successfully, but if you then try to delete the _same exact file_, you get the error message?  (And you're sure the rename actually "sticks"? - i.e. check on disk to make sure the file was renamed for real).

If so, that's pretty darn weird!  Is your project on an unusual filesystem?  I think you can type `df -T <path_to_file>` to find out what filesystem the impossible-to-delete file lives on...

If you _create_ a new file using Brackets in the same folder, can you delete that file?  If so, I'd be suspicious of some kind of weird permissions issue too... You can use `ls -l` to compare permissions of the deletable file vs. impossible-to-delete ones.

**Comment by millerbarros** [comment]
The partition where my projects are is a NTFS partition mounted to `/media/dados` when I excuto the command `df -T /media/dados/<path_to_project_folder>`, I get the following return:

```
File Sys.      Type      Mount
/dev/sda4      fuseblk   /media/dados
```

Apparently "fuseblk" makes reference to a ntfs partition mounted by the "mount" command, or am I mistaken?

In any file of any project that is within that partition I get this error, even if I have created this file directly by Brackets sidebar, but if I use the system partition (ext4) files are deleted without problems.

Unfortunately I do not how to format that partition to another file system, it is a partition with 900 gb of files coming from when I used windows. Is there anything I can do to resolve this error?

**Comment by peterflynn** [comment]
@millerbarros It sounds like moving to trash might not work on NTFS partitions unless they're mounted a certain way -- see http://ubuntuforums.org/showthread.php?t=1424167.  It's also worth checking whether the trash/recycle folder on that partition has the right permissions -- see http://forums.linuxmint.com/viewtopic.php?f=90&t=66535.

Ideally, if moving to trash fails, Brackets should show a warning and offer to hard-delete the file instead.  Currently however, if `g_file_trash()` fails then the Brackets command gives up.

**Comment by millerbarros** [comment]
You were right, the problem was related to trash. I tried to delete to trash a file on this partition using Nemo and got a message saying it is not possible to delete to trash and asked if I would like to delete directly.
Researched and found others with the same problem as me, which was configured to automatically mount using the default OS installation options. So I changed the mount options that partition and now everything works perfectly (including delete in Brackets).

My solution was as follows:

I had the following setting in /etc/fstab file:
```
/dev/disk/by-uuid/XXXXXX /media/dados auto nosuid,nodev,nofail,x-gvfs-show 0 0
```
And changed to:
```
/dev/disk/by-uuid/XXXXXX /media/dados ntfs defauls,nls=utf8,umask=000,uid=1000\0400\0400 0 0
```
Reboot the Machine!!!

Problem solved, thanks for helping @peterflynn. The Breckets is impressive.
